### PR TITLE
Coin contract need to validate some field in the request

### PR DESCRIPTION
**Pull request by atchapcyp**
Since some method in coin contract did't check the value before trying to unmarshal.
So It is possible to unmarshal `nil` variable.
To prevent this we should validate variable before unmarshal.

In order to do that we need ...
- Check the variable before unmarshal. 
- This change needs to be enabled by feature flag `coin:v1.2`
- Add coin contract unit test for flag `coin:v1.2` is enabled.

ref : #902 
- [ ] I added unit tests for any code that added
- [ ] I updated the CHANGELOG.md 
- [ ] All IP is original and not copied from another source
- [ ] I assign all copyright to Loom Network for the code in the pull request

### CODE REVIEW

File: builtin/plugins/coin/coin.go [hunk 1]
```diff
@@ -318,6 +330,9 @@ func (c *Coin) Allowance(
 	ctx contract.StaticContext,
 	req *AllowanceRequest,
 ) (*AllowanceResponse, error) {
+	if ctx.FeatureEnabled(loomchain.CoinVersion1_2Feature, false) && (req.Spender == nil || req.Owner == nil) {
```
Comment (pathornteng) [inline builtin/plugins/coin/coin.go]: Like BalanceOf, Allowance does not change state of the chain, so the feature flag check is unnecessary. All methods that use `contract.StaticContext` are quite safe to be changed without a feature flag.

File: builtin/plugins/coin/coin.go [hunk 2]
```diff
@@ -233,6 +234,10 @@ func (c *Coin) BalanceOf(
 	ctx contract.StaticContext,
 	req *BalanceOfRequest,
 ) (*BalanceOfResponse, error) {
+
```
Comment (enlight) [inline builtin/plugins/coin/coin.go]: Get rid of the blank line

File: builtin/plugins/coin/coin.go [hunk 3]
```diff
@@ -332,6 +346,12 @@ func (c *Coin) Allowance(
 }
 
 func (c *Coin) TransferFrom(ctx contract.Context, req *TransferFromRequest) error {
+
```
Comment (enlight) [inline builtin/plugins/coin/coin.go]: Get rid of the blank line

File: builtin/plugins/coin/coin.go [hunk 4]
```diff
@@ -417,6 +434,9 @@ func loadAccount(
 }
 
 func saveAccount(ctx contract.Context, acct *Account) error {
+	if ctx.FeatureEnabled(loomchain.CoinVersion1_2Feature, false) && (acct.Owner == nil || acct.Balance == nil) {
+		return ErrInvalidRequest
```
Comment (enlight) [inline builtin/plugins/coin/coin.go]: This isn't a contract method, so returning `ErrInvalidRequest` doesn't make sense in this context, you should move the check to the caller of `saveAccount`

File: builtin/plugins/coin/coin.go [hunk 5]
```diff
@@ -441,6 +461,11 @@ func loadAllowance(
 }
 
 func saveAllowance(ctx contract.Context, allow *Allowance) error {
+	if ctx.FeatureEnabled(loomchain.CoinVersion1_2Feature, false) {
+		if allow.Owner == nil || allow.Spender == nil || allow.Amount == nil {
+			return ErrInvalidRequest
```
Comment (enlight) [inline builtin/plugins/coin/coin.go]: See comment on `saveAccount`, same thing applies here.

**Pull request merged**